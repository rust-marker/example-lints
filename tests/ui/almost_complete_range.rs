//@no-rustfix
#[allow(clippy::manual_is_ascii_check)]

fn main() {
    let x = 0;
    let c = 'x';

    // Lint
    let _ = (b'a'..b'z').contains(&x);
    let _ = (b'A'..b'Z').contains(&x);
    let _ = (b'0'..b'9').contains(&x);
    let _ = ('a'..'z').contains(&c);
    let _ = ('A'..'Z').contains(&c);
    let _ = ('0'..'9').contains(&c);

    // Don't lint (inclusive)
    let _ = (b'a'..=b'z').contains(&x);
    let _ = (b'A'..=b'Z').contains(&x);
    let _ = (b'0'..=b'9').contains(&x);
    let _ = ('a'..='z').contains(&c);
    let _ = ('A'..='Z').contains(&c);
    let _ = ('0'..='9').contains(&c);

    let b: i32 = 32;

    // Don't lint (something else)
    let _ = (b'a'..).contains(&x);
    let _ = (..=b'Z').contains(&x);
    let _ = (256..7272).contains(&b);
    let _ = (0..1).contains(&x);
    let _ = (0..9).contains(&x);
}
