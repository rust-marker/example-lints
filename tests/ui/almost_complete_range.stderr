warning: almost complete ascii range
 --> $DIR/almost_complete_range.rs:8:13
  |
8 |     let _ = (b'a'..b'z').contains(&x);
  |             ^^^^^^^^^^^^ help: try: `b'a'..=b'a'`
  |
  = note: `#[warn(marker::almost_complete_range)]` on by default

warning: almost complete ascii range
 --> $DIR/almost_complete_range.rs:9:13
  |
9 |     let _ = (b'A'..b'Z').contains(&x);
  |             ^^^^^^^^^^^^ help: try: `b'A'..=b'A'`

warning: almost complete ascii range
  --> $DIR/almost_complete_range.rs:10:13
   |
10 |     let _ = (b'0'..b'9').contains(&x);
   |             ^^^^^^^^^^^^ help: try: `b'0'..=b'0'`

warning: almost complete ascii range
  --> $DIR/almost_complete_range.rs:11:13
   |
11 |     let _ = ('a'..'z').contains(&c);
   |             ^^^^^^^^^^ help: try: `'a'..='a'`

warning: almost complete ascii range
  --> $DIR/almost_complete_range.rs:12:13
   |
12 |     let _ = ('A'..'Z').contains(&c);
   |             ^^^^^^^^^^ help: try: `'A'..='A'`

warning: almost complete ascii range
  --> $DIR/almost_complete_range.rs:13:13
   |
13 |     let _ = ('0'..'9').contains(&c);
   |             ^^^^^^^^^^ help: try: `'0'..='0'`

warning: 6 warnings emitted

